### PR TITLE
Update XGBoost API and examples

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
- load_data only requires return dmat_train and dmat_valid
- generate one app for each site (for lr_scale)

### CODE REVIEW

**Review by ZiyueXu77 — APPROVED** [review]
LGTM

File: examples/xgboost/utils/prepare_job_config.py [hunk 1]
```diff
@@ -26,8 +26,9 @@
 
 def job_config_args_parser():
     parser = argparse.ArgumentParser(description="generate train configs for HIGGS dataset")
-    parser.add_argument("--data_split_path", type=str, default="./data_splits", help="Path to data split folder")
+    parser.add_argument("--data_root", type=str, default="/tmp/nvflare/xgboost_higgs_dataset", help="Path to dataset")
```
Comment (ZiyueXu77) [inline examples/xgboost/utils/prepare_job_config.py:29]: Path to dataset config json files for each client? (dataset path is "~/dataset/HIGGS.csv")

Comment (YuanTingHsieh) [reply examples/xgboost/utils/prepare_job_config.py]: thanks

**Review by ZiyueXu77 — APPROVED** [review]

**Pull request merged**